common/lib/xmodule/xmodule/course_module.py

```diff
@@ -741,6 +741,31 @@ class CourseFields(object):
         scope=Scope.settings,
     )
 
+    specialization_title_en = String(
+        display_name="Specialization Title",
+        help=_("Enter the title for the  specialization which this "
+               "course belongs to."),
+        default="",
+        scope=Scope.settings
+    )
+
+    specialization_title_ar = String(
+        display_name="Specialization Arabic Title",
+        help=_("Enter the arabic title for the specialization which "
+               "this course belongs to."),
+        default="",
+        scope=Scope.settings
+    )
+
+    specialization_link = String(
+        display_name="Specialization Link",
+        help=_("Enter the url linking to the specialization's "
+               "learner page"),
+        default="",
+        scope=Scope.settings
```

Comment:
This is not generic, and very manual, I think the best way to get the link is through providing an ID here, then using settings variable of PROGS_PROGRAM_LMS_LINK, a url can be created using the ID and url schema of progs. Let me know your thoughts @saadaltabari

Reply:
Yeah, I thought of that way at first but it would require adding programs platform's urls to the edx settings and after I shared this with @devalih he advised against having edx dependent on progs in general and I thought it will be more of a hassle to have to make sure progs urls are configured correctly on all environments